### DISCUSSION TITLE
Sftp polling until remote file available

### DISCUSSION THREAD

**Discussion by af6140** [spring-projects/spring-integration#3849] [Q&A] [answered]
Hi trying to figure out how to wait until remote sftp file is available with a timeout. Was thinking using inbound channel adapter with message gateway , but not sure how it can be achieved.

**Comment by oxcafedead** [comment, marked as answer]
Hi @af6140
Please take a look at https://docs.spring.io/spring-integration/reference/html/sftp.html#configuring-with-java-configuration

Specifically this bean definition:
```java
    @Bean
    @InboundChannelAdapter(channel = "sftpChannel", poller = @Poller(fixedDelay = "5000"))
    public MessageSource<File> sftpMessageSource() {
        SftpInboundFileSynchronizingMessageSource source =
                new SftpInboundFileSynchronizingMessageSource(sftpInboundFileSynchronizer());
        source.setLocalDirectory(new File("sftp-inbound"));
        source.setAutoCreateLocalDirectory(true);
        source.setLocalFilter(new AcceptOnceFileListFilter<File>());
        source.setMaxFetchSize(1);
        return source;
    }
```

It should do what you have described if I understand your request correctly.

**Reply by artembilan** [reply]
Moved this to discussion, since the question does not fall to `bug` neither `feature request` category.

Please, see if general EIP theory may help you somehow to determine what algorithm does fit to your requirements: https://www.enterpriseintegrationpatterns.com/patterns/messaging/.

In two works, @oxcafedead , has answered your question, but in addition I'd like to share also this [part of the docs](https://docs.spring.io/spring-integration/docs/current/reference/html/polling-consumer.html#polling-consumer) to make it clear what type of components is for this type of tasks.

**Reply by artembilan** [reply]
Also: please, consider to ask questions in the future on StackOverflow.
GH issues are only for bugs and feature requests.
GH discussions have limited community reachability.

Thanks
